src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java

```diff
@@ -1172,6 +1172,26 @@ protected void writeConfiguration() throws IOException {
     }
 
     /**
+     * Send message to webapp to refresh SearcherManagers for given projects.
+     * This is used for partial reindex.
+     *
+     * @param subFiles list of directories to refresh corresponding SearcherManagers
+     * @param host the host address to receive the configuration
+     * @param port the port to use on the host
+     * @throws IOException if an error occurs
+     */
+    public void signalTorefreshSearcherManagers(List<String> subFiles, String host, int port) throws IOException {
+        Message m = Message.createMessage("refresh");
+        m.setText("refresh");
```

Comment:
Is the text necessary? Add a validation for text field? And possibly create a test for this message testing the validation.